net/splitlistener.go

```diff
@@ -288,6 +307,19 @@ func (l *MultiplexingListener) Accept() (net.Conn, error) {
 			return nil, net.ErrClosed
 
 		default:
+			if in.conn == nil {
+				return in.conn, in.err
+			}
+
+			switch t := in.conn.(type) {
+			case *tls.Conn:
+				return in.conn, in.err
+			case *protocol.Conn:
+				if l.nativeConns {
```

Comment:
The key is further up in the file. Prior to this change we would _only ever_ return a `*tls.Conn` because what would come in would always be a `*protocol.Conn` and we'd strip the `*tls.Conn` out and push that down the channel. So further up in the file we're instead now sending the `*protocol.Conn` down the wire, but for many/most cases you really want to send along the `*tls.Conn`...for instance, there is code in gRPC that expects that a connection coming in is a `*tls.Conn`.

So this basically keeps the status quo for the normal case but allows you to have the native type returned if you want. As a side benefit, if you're using a multiplexing listener outside of the context of a split listener, this also means that anything that implements `net.Conn` will actually be able to be serviced by this type now.

Reply:
To be clear -- we would only ever return a `*tls.Conn` if you were using this with `SplitListener` -- this lib wouldn't care either way and would just return what was there.

I guess whether this is a _breaking_ change comes down to if you were using `*protocol.Conn` directly with this before, and now you'd have to specify to use native conns. Boundary wasn't, so I feel confident in saying that nothing else was either.

The other alternative would be to set it on `SplitListener` but then you'd have to deal with it for all listeners you got out of there, which I don't think is ideal.

Reply:
Done -- see if the updated logic makes more sense!